### PR TITLE
http Headers: fix misc issues

### PR DESCRIPTION
**Pull request by vicb**
/cc @robwormald

### CODE REVIEW

**Comment by chuckjaz** [comment]
Please change commit messages to begin with "refactor(http):", "fix(http):" respectively.

**Comment by francosioquim** [comment]
/cc @robwormald
You can view, comment on, or merge this pull request online at:

   https://github.com/angular/angular/pull/12106

-- Commit Summary --

   * refactor(Headers): move one test inside the main describe
   * fix(Headers): append should append to the list
   * fix(Headers): fix initialization from Headers and Object

-- File Changes --

     M modules/@angular/http/src/headers.ts (29)
     M modules/@angular/http/test/headers_spec.ts (53)

-- Patch Links --

https://github.com/angular/angular/pull/12106.patch
https://github.com/angular/angular/pull/12106.diff

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/angular/angular/pull/12106

File: modules/@angular/http/src/headers.ts [hunk 1]
```diff
@@ -49,19 +49,16 @@ export class Headers {
...
       });
       return;
     }
 
     Object.keys(headers).forEach((name: string) => {
-      const value = headers[name];
-      const lcName = name.toLowerCase();
-      this._headers.set(lcName, Array.isArray(value) ? value : [value]);
-      this.mayBeSetNormalizedName(name);
+      const values: string[] = Array.isArray(headers[name]) ? headers[name] : [headers[name]];
```
Comment (robwormald) [inline modules/@angular/http/src/headers.ts]: this is ES6, correct? would require a polyfill - do we use it elsewhere / expect it to be available?

**Comment by robwormald** [comment]
LGTM

**Comment by tbosch** [comment]
Landed in master.

**Pull request closed without merging**